src/internal.d.ts

```diff
@@ -70,12 +70,13 @@ export interface VNode<P = {}> extends preact.VNode<P> {
 	/**
 	 * The [first (for Fragments)] DOM child of a VNode
 	 */
-	_dom: PreactElement | Text | null;
+	_dom: PreactElement | null;
 	/**
 	 * The last dom child of a Fragment, or components that return a Fragment
 	 */
-	_nextDom: PreactElement | Text | null;
+	_nextDom: PreactElement | null;
 	_component: Component | null;
+	_hydrating: boolean | null;
```

Comment:
Perhaps this wouldn't compress as well since we deal with nulls way more than booleans, but just leaving a note for myself (cuz I'm curious) to explore pros/cons of just making this a normal boolean that is initialized to `false` and flipped between `true`/`false` as appropriate.

Reply:
@andrewiggins this needs to be a tri-state though:
- `true`: the vnode has DOM from bailout during hydration
- `false`: the vnode has DOM from bailout during initial render (new tree creation)
- `null`: the vnode has no DOM from bailout

We could use an int though, that's probably much better for perf.

Reply:
Oh, I missed that `false` case. Makes sense